src/common/pick_address.cc

```diff
@@ -45,9 +45,9 @@ bool matches_with_name(const ifaddrs& ifa, const std::string& if_name)
   return if_name.compare(ifa.ifa_name) == 0;
 }
 
-bool is_addr_allowed(const ifaddrs& ifa)
+bool is_addr_allowed(const ifaddrs& ifa, bool exclude_lo_iface)
 {
-  if (strcmp(ifa.ifa_name, "lo") == 0) {
+  if (exclude_lo_iface && strcmp(ifa.ifa_name, "lo") == 0) {
```

Comment:
what about ``lo:0`` etc?  should this check for ``lo`` or prefix ``lo:``?

Reply:
it is checked at line 53.

Reply:
i think i am trying to address the issue fixed by #38925. in which, `lo` is allowed to be bound by daemons, while the `lo:` interfaces are not. and in the latest master, both `lo` and `lo:` are not allowed, so this option is introduced for the very use case of https://tracker.ceph.com/issues/48893.